androidApp/src/main/java/com/example/tasky/android/common/viewmodel/MainViewModel.kt

```diff
@@ -15,8 +19,29 @@ import kotlinx.coroutines.flow.stateIn
 class MainViewModel(
     konnection: Konnection,
     private val agendaRepository: IAgendaRepository,
-) : ViewModel() {
-    val isLoginStateFlow = isLoginFlow.stateIn(scope = viewModelScope, started = SharingStarted.Eagerly, initialValue = false)
+    application: Application,
+) : AndroidViewModel(application) {
+    val isLoginStateFlow =
+        isLoginFlow
+            .onEach {
+                if (it) {
+                    val receiver = ComponentName(application, BootReceiver::class.java)
+
+                    application.packageManager.setComponentEnabledSetting(
+                        receiver,
+                        PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
+                        PackageManager.DONT_KILL_APP,
+                    )
+                } else {
+                    val receiver = ComponentName(application, BootReceiver::class.java)
+
+                    application.packageManager.setComponentEnabledSetting(
+                        receiver,
+                        PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
+                        PackageManager.DONT_KILL_APP,
```

Comment:
Considering that this requires Context, could it not make more sense to place this behind an interface that your VM can depend on?